### PR TITLE
Auto-commit clang format

### LINKED ISSUE #1457: Auto-commit clang format
**Is your feature request related to a problem? Please describe.**

I know we spoke about this but I forgot the limitation, so let's persist this discussion in an issue. We all have forgetten to clang format at one point.

**Describe the solution you'd like**

It would be really nice if the CI would just do it for us.

Reading here
https://stackoverflow.com/questions/57921401/push-to-origin-from-github-action/58393457#58393457
this should be possible with the `run` command. And then it brings up this question of who should be the author but then it could just be the bot itself
```
git config user.name 'github-actions[bot]'
git config user.email 'github-actions[bot]@users.noreply.github.com'
```

### PR DESCRIPTION
**Pull request by tvami**
I am updating _ldmx-sw_, here are the details.

### What are the issues that this addresses?
Resolves https://github.com/LDMX-Software/ldmx-sw/issues/1457

## Check List
- [x] I successfully compiled _ldmx-sw_ with my developments
- [x] I ran my developments and the following shows that they are successful.

The test happened on GitHub:
 * If there is a change to apply, the bot indeed did it
 * If no change to apply, it didnt do anything

### CODE REVIEW

**Comment by tvami** [comment]
Wow look at that!

Screenshot 2024-09-17 at 13 33 50

**Comment by tvami** [comment]
> 1 cancelled and 1 successful checks

I cancelled the basic test since it's not relevant to the change, the other one tests everything we need

File: .github/workflows/format-check.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Check Format
+name: Check/apply clang-format
```
Comment (tomeichlersmith) [inline .github/workflows/format-check.yml]:
```suggestion
name: apply clang-format
```

**Review by tomeichlersmith — COMMENTED** [review]
just a cosmetic change to make the name appearing the GitHub messages a bit easier to parse. (GitHub uses the `/` character for separating the workflow name from other details.)

File: .github/workflows/format-check.yml (hunk 1 above)
Comment (tvami) [reply .github/workflows/format-check.yml]: Ah ok, I just wanted to have the case when the bot will not make a commit, but ok I'll remove the / now

**Review by tomeichlersmith — APPROVED** [review]

**Review by danyi211 — APPROVED** [review]

**Pull request merged**